axis/src/commonMain/kotlin/ContinuousAxis.kt

```diff
@@ -0,0 +1,221 @@
+
+@JvmName("axisBottomLocalDateTime")
+public fun axisBottom(scale: ContinuousScale<LocalDateTime, Float>): ContinuousAxis<LocalDateTime> = ContinuousAxis(Bottom, scale, LocalDateTimeTicker)
+
+@JvmName("axisLeftInstant")
+public fun axisLeft(scale: ContinuousScale<Instant, Float>): ContinuousAxis<Instant> = ContinuousAxis(Left, scale, InstantTicker)
+
+@JvmName("axisTopInstant")
+public fun axisTop(scale: ContinuousScale<Instant, Float>): ContinuousAxis<Instant> = ContinuousAxis(Top, scale, InstantTicker)
+
+@JvmName("axisRightInstant")
+public fun axisRight(scale: ContinuousScale<Instant, Float>): ContinuousAxis<Instant> = ContinuousAxis(Right, scale, InstantTicker)
+
+@JvmName("axisBottomInstant")
+public fun axisBottom(scale: ContinuousScale<Instant, Float>): ContinuousAxis<Instant> = ContinuousAxis(Bottom, scale, InstantTicker)
+
+public fun Selection<*, *>.call(axis: ContinuousAxis<*>): Unit = axis.applySelection(this)
+
+// TODO: Axis should support scales other than ContinuousScale, but since we don't have other scales yet...
+public class ContinuousAxis<D : Comparable<D>> internal constructor(
+    private val edge: Edge,
+    private val scale: ContinuousScale<D, Float>,
+    private val ticker: Ticker<D>,
+) {
+    public var tickSizeInner: Float = 6f
+    public var tickSizeOuter: Float = 6f
+    public var tickPadding: Float = 3f
+    public var tickCount: Int = 5
+
+    public var font: Font = Font(sansSerif)
+    public var textSize: Float = 14f
+    public var textColor: Color = black
+    public var lineWidth: Float = 1f
+    public var lineColor: Color = black
+    public var formatter: (D) -> String = { it.toString() }
+
+    private val k = if (edge == Top || edge == Left) -1 else 1
+    private val isVertical = edge == Left || edge == Right
+
+    public fun applySelection(selection: Selection<*, *>) {
+        val values = ticker.ticks(scale.domain.minOf { it }, scale.domain.maxOf { it }, tickCount)
+        val spacing = max(tickSizeInner, 0f) + tickPadding
+
+        val range = scale.range
+        val range0 = range.first()
+        val range1 = range.last()
+
+        val path = selection.selectAll(PathElement.withKind("domain"))
+            .data(listOf(null))
+
+        val tick = selection.selectAll(TransformElement.withKind("tick"))
+            .keyedData(values) { (d) -> (d as? D)?.let(scale::scale) }
+            .order()
+        val tickEnter = tick.enter.append(TransformElement)
+            .each { kind = "tick" }
+        val tickExit = tick.exit
+
+        val line = tick.select(LineElement)
+
+        val text = tick.select(TextElement)
+
+        val linePaint = Paint.Stroke(lineColor, lineWidth)
+        val pathMerge = path.merge(
+            path.enter.insert(PathElement, Element.withKind("tick"))
+                .each { kind = "domain" }
+        ).each { paint = linePaint }
```

Comment:
Apologies that you're having to teach me how D3 works, but don't quite follow how this works.
We `selectAll` of kind `"domain"` on line 123 _before_ we've even `insert`ed the elements (line 139)?

Reply:
> Apologies that you're having to teach me how D3 works

No worries! I realize it's totally alien outside of the web space.

> We selectAll of kind "domain" on line 123 before we've even inserted the elements (line 139)?

Correct. When talking about the selection process, including `selectAll`, there's really two cases I keep in my head: _first run_ behavior, and all future runs.

On the first run, `selectAll` will return a `Selection` with a single `Group` containing zero `Element`s. This is fine. When we associate `.data` with that selection, the empty list simply means that all items in the data will be in the enter sub-selection, and then `.append` or `.insert` will add an element for each point of data.

On future runs, `selectAll` will return a `Selection` whose `Group` actually contains some `Elements`. Depending on the `.data` passed in, we'll potentially have elements in the update (directly returned) selection, the enter sub-selection (for new items), and the exit sub-selection (for items that no longer exist). Note that with `.data` you can only have one of enter and exit, but with `.keyedData` you can have both at once.

Reply:
Also worth noting that this file handles things in a much more complicated way than I would have written them naturally. I preserved the original logic because it'll be useful once we add true animation support, but often `.join` is an easier paradigm than thinking about update/enter/exit as separate selections.

Reply:
As always, your explanations are phenomenal! Makes way more sense now! TY! TY!